ink/crates/phat_rollup_anchor_ink/src/traits/message_queue.rs

```diff
@@ -0,0 +1,36 @@
+use ink::prelude::vec::Vec;
+pub use kv_session::traits::QueueIndex;
+use scale::{Decode, Encode};
+
+#[derive(Debug, Eq, PartialEq, scale::Encode, scale::Decode)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub enum MessageQueueError {
+    InvalidPopTarget,
+    FailedToDecode,
+}
+
+#[openbrush::trait_definition]
+pub trait MessageQueue {
+    fn _push_message<M: Encode>(&mut self, data: &M) -> Result<QueueIndex, MessageQueueError>;
```

Comment:
The leading `_` in Rust is used to suppress compiler warnings for unused items.
If the aim is to decouple requirements for implementers, you can break down methods into separate traits.

Reply:
I used the prefix `_`to say the method is not callable (without #[ink(message)] ) but maybe it is not a good convention for the Traits. For the Impl, I think we can still use the prefix underscorse `_` to say the method is internal. (OpenBrush uses it for [internal functions](https://github.com/Brushfam/openbrush-contracts/blob/44a7d7ba6e9afbffac62116cba4edf84ae39f0ce/example_project_structure/contracts/loan/lib.rs#L109)).
So probably for all internal functions (not callable function and not defined in the Trait), I can still to use the prefix `_` but I have to find another convention for not callable function defined in the Trait.

Reply:
If we want then internal only, we should break down methods into separate traits, and decorate internal onces with `pub(crate) trait`

Reply:
> I think we can still use the prefix underscorse `_` to say the method is internal. (OpenBrush uses it for [internal functions](https://github.com/Brushfam/openbrush-contracts/blob/44a7d7ba6e9afbffac62116cba4edf84ae39f0ce/example_project_structure/contracts/loan/lib.rs#L109)).

It's not recommended to use the prefix `_` to say the method is internal since Rust already has built-in visibility control(`pub`). Using `_` has the side effect that suppressing the compiler warnings if the method is unused.